azafea/event_processors/endless/metrics/v3/model/views.py

```diff
@@ -0,0 +1,678 @@
+        ParentalControlsBlockedFlatpakInstallV3.occured_at,
+        ParentalControlsBlockedFlatpakInstallV3.os_version,
+        ParentalControlsBlockedFlatpakInstallV3.app,
+        Channel.image_id,
+        Channel.site,
+        Channel.dual_boot,
+        Channel.live
+    ).select_from(
+        ParentalControlsBlockedFlatpakInstallV3
+    ).join(
+        Channel
+    ).union(
+        ParentalControlsBlockedFlatpakInstallViewV2
+    )
+
+
+class ParentalControlsBlockedFlatpakRunView(View):
+    __tablename__ = 'parental_controls_blocked_flatpak_run_view'
+
+    __query__ = DbSession().query(
+        ParentalControlsBlockedFlatpakRunV3.occured_at,
+        ParentalControlsBlockedFlatpakRunV3.os_version,
+        ParentalControlsBlockedFlatpakRunV3.app,
+        Channel.image_id,
+        Channel.site,
+        Channel.dual_boot,
+        Channel.live
+    ).select_from(
+        ParentalControlsBlockedFlatpakRunV3
+    ).join(
+        Channel
+    ).union(
+        ParentalControlsBlockedFlatpakRunViewV2
+    )
+
+
+class ParentalControlsChangedView(View):
+    __tablename__ = 'parental_controls_changed_view'
+
+    __query__ = DbSession().query(
+        ParentalControlsChangedV3.occured_at,
+        ParentalControlsChangedV3.os_version,
+        ParentalControlsChangedV3.app_filter_is_whitelist,
+        ParentalControlsChangedV3.app_filter,
+        ParentalControlsChangedV3.oars_filter,
+        ParentalControlsChangedV3.allow_user_installation,
+        ParentalControlsChangedV3.allow_system_installation,
+        ParentalControlsChangedV3.is_administrator,
+        ParentalControlsChangedV3.is_initial_setup,
+        Channel.image_id,
+        Channel.site,
+        Channel.dual_boot,
+        Channel.live
+    ).select_from(
+        ParentalControlsChangedV3
+    ).join(
+        Channel
+    ).union(
+        ParentalControlsChangedViewV2
+    )
+
+
+class ParentalControlsEnabledView(View):
+    __tablename__ = 'parental_controls_enabled_view'
+
+    __query__ = DbSession().query(
+        ParentalControlsEnabledV3.occured_at,
+        ParentalControlsEnabledV3.os_version,
+        ParentalControlsEnabledV3.enabled,
+        Channel.image_id,
+        Channel.site,
+        Channel.dual_boot,
+        Channel.live
+    ).select_from(
+        ParentalControlsEnabledV3
+    ).join(
+        Channel
+    ).union(
+        ParentalControlsEnabledViewV2
+    )
+
+
+class ProgramDumpedCoreView(View):
+    __tablename__ = 'program_dumped_core_view'
+
+    __query__ = DbSession().query(
+        ProgramDumpedCoreV3.occured_at,
+        ProgramDumpedCoreV3.os_version,
+        ProgramDumpedCoreV3.info,
+        Channel.image_id,
+        Channel.site,
+        Channel.dual_boot,
+        Channel.live
+    ).select_from(
+        ProgramDumpedCoreV3
+    ).join(
+        Channel
+    ).union(
+        ProgramDumpedCoreViewV2
+    )
+
+
+class UpdaterFailureView(View):
+    __tablename__ = 'updater_failure_view'
```

Comment:
I'm not very keen on the "view" suffix here. Similarly the `_view_v2` suffix is a bit opaque.

Perhaps…

- `updater_failure`: real V2 data
- `updater_failure_v3`: real V3 data
- `updater_failure_v2_as_v3`: materialized view of V2 data in V3 format
- `updater_failure_all`: combined view of v2 & v3 data

I'm not all that keen on those names either and have asked on #os for other suggestions :)

(Here I am about to overcomplicate things, probably, but… we could use a separate [postgresql schema](https://www.postgresql.org/docs/current/ddl-schemas.html) to namespace these views. All the current views and tables are in `public`. Maybe a `combined` schema containing `updater_failure` (which is the combined view of v2 and v3 data) and a `v2_as_v3` schema containing the materialized view? As I type this, I think this would actually be more confusing.)

Reply:
@dbnicholson thinks that either `updater_failure_v2_as_v3` or `updater_failure_view_v2` is fine for the materialized view, and `updater_failure_all` is OK for the combined view. From our Slack:

> i think that naming makes sense, although i don't believe A actually has to change. the events did come from the v2 protocol. the fact that their being views in the v3 format is IMO an implementation detail of v3 being the current format. i don't believe you want to keep viewing the v2 events in v2 format